### Matched Pair design

Matched pair design experiments are ususally used when the volume of observations or individuals is fairly low, the concern for bias is great, and the cost per obervation is high by comparison. Mathced pairing sets up treatment and control groups for an experiment by matching on unit by unit basis using a set of control variables. The goal is to create two sets of units tht are as similar as possible to each other, so the we can confidently say that any changes during the test are due to the introduction of experiment variable. A mathematicla alogorithm is used to create the treatmemnt and control groups. Where for each treatment unit, one or more control units is selected.

###  Selection of treatment units

1. Identify  outliers

2. Decide number of treatment units

3. Randomly select treatment units

###  Selection of control store

Matched pairing is where each treatment store is matched to a control store that has very similar measure for each of the control variables.

### New Product experiment

We choose three control variables, sales volume, number of products sold and state where the store is located.

In [73]:
import pandas as pd
import numpy as np
%matplotlib inline
import matplotlib.pyplot as plt
import seaborn as sns
import datetime as dt
from statsmodels.stats.power import TTestIndPower
import statsmodels.api as sm
from pandas import get_dummies
import category_encoders as ce
from sklearn.preprocessing import LabelEncoder
from sklearn.linear_model import LogisticRegression
from scipy.stats import linregress
from sklearn.linear_model import LogisticRegression
import ggplot
from pandas.lib import Timestamp
from scipy import stats

In [74]:
df = pd.read_csv("newproductcontroldata.csv")

In [75]:
df.head()

Store_ID        City State  Zip_Code  Category_Sales  Product_Count   Size
0         1   ALABASTER    AL     35007           18.88              1  Large
1         2  BIRMINGHAM    AL     35209        44125.66              6  Large
2         3     DECATUR    AL     35601        46627.92              5  Large
3         4  HUNTSVILLE    AL     35806        26658.48              6  Large
4         5      MOBILE    AL     36606         1863.60              3  Large

In [76]:
df.shape

(7008, 7)

### Steps for Selecting Control Units

The three control variables we want to match on are Sales Volume, Number of Products, and State.

First, let’s filter the data so that the state matches.

Sort the data so that sales is in ascending order. Search for Store ID 111 in the list.

Determine which store is the closest match to it

In [77]:
print(df[df['Store_ID'] == 111])

     Store_ID    City State  Zip_Code  Category_Sales  Product_Count   Size
110       111  Irvine    CA     92618         9556.14              5  Large


### Review for each potential

In [78]:
df[(df.Product_Count == 5) & (df.State == 'CA') & (df.Category_Sales >= 9200) & (df.Category_Sales <= 9650)]

Store_ID        City State  Zip_Code  Category_Sales  Product_Count  \
110        111      Irvine    CA     92618         9556.14              5   
662        663  Costa Mesa    CA     92626         9480.82              5   
2658      2659   Mira Loma    CA     91752         9305.58              5   
5034      5035   Roseville    CA     95661         9602.96              5   
5375      5376    Pasadena    CA     91101         9405.92              5   

        Size  
110    Large  
662    Large  
2658   Large  
5034   Large  
5375  Medium

In [79]:
df[(df.Product_Count == 6) & (df.State == 'CA') & (df.Category_Sales >= 9000) & (df.Category_Sales <= 10000)]

Store_ID         City State  Zip_Code  Category_Sales  Product_Count  \
2689      2690     Pasadena    CA     91107         9655.01              6   
3497      3498  Santa Maria    CA     93455         9532.24              6   
5332      5333     Pasadena    CA     91101         9818.00              6   

        Size  
2689   Large  
3497   Large  
5332  Medium

### Analysis

From pre processing above we have find out two store closest to store 111 are store 3498 fo closest sales but Product count is diffrent and store 5053 is also closest

#### Experiment

For the new product introduction experiment. your manager is assigend a set of 10 treatment stores in 1 state where the cherry flavor of the product will be sold. We will matching 2 control stores to each of the stores for set of treatment stores for cherry flavor.

In [95]:
Treatment = df[(df.State == 'CA')]

In [96]:
df1 = pd.DataFrame(Treatment)

In [97]:
candy_treatment = df1.sample(10)

In [98]:
candy_treatment.head()

Store_ID             City State  Zip_Code  Category_Sales  \
4799      4800           Upland    CA     91784        23869.61   
3071      3072  Manhattan Beach    CA     90266        17155.60   
6033      6034          Anaheim    CA     92805        54942.34   
3951      3952           Sanger    CA     93657         3104.32   
1351      1352          Lincoln    CA     95648        32177.44   

      Product_Count    Size  
4799              6   Large  
3071              5   Large  
6033              6   Small  
3951              4  Medium  
1351              5   Large